### ISSUE TITLE
System.ArgumentOutOfRangeException thrown if verbosity has an incorrect value

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#7677]
## Steps to reproduce
I used `debug` instead `d` for `-v` and instead of error saying that the used verbosity is not valid I got this:
```
dotnet restore -v debug
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collectio
n.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException()
   at System.SZArrayHelper.get_Item[T](Int32 index)
   at System.Linq.Enumerable.ElementAt[TSource](IEnumerable`1 source, Int32 index)
   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.LogVerbosityForAllTopLevelCommand(ICollection`1 result, ParseResult
 parseResult, String topLevelCommandName)
   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.Filter(Object objectToFilter)
   at Microsoft.DotNet.Cli.Utils.TelemetryEventEntry.SendFiltered(Object o)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```
## Expected  behavior
A meaningful error message

## Actual behavior
ArgumentOutOfRangeException

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.1.0-preview1-007042)

Product Information:
 Version:            2.1.0-preview1-007042
 Commit SHA-1 hash:  e94a82c9a3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\XXX\.dotnet\x64\sdk\2.1.0-preview1-007042\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

**Comment by randomizedthinking** [comment]
Just encountered the same issue today on dotnet SDK `2.0.0`.

**Comment by mikkelbu** [comment]
I think this will be solved by #7636

**Comment by wli3** [comment]
Yes, this issue is fixed

**Comment by lucasgalone** [comment]
How was it solved? help me?

**Comment by wli3** [comment]
@lucasgalone It is fixed in 2.1.300
